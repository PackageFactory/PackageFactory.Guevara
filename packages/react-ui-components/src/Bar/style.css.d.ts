export interface IStyleCss {
  'bar': string;
  'bar--top': string;
  'bar--bottom': string;
}

export const locals: IStyleCss;
