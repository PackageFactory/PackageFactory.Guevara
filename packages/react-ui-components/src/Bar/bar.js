import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import mergeClassNames from 'classnames';

class Bar extends PureComponent {
    static propTypes = {
        /**
         * This prop controls the vertical positioning of the Bar.
         */
        position: PropTypes.oneOf(['top', 'bottom']).isRequired,

        /**
         * An optional css theme to be injected.
         */
        theme: PropTypes.shape({
            'bar': PropTypes.string,
            'bar--top': PropTypes.string,
            'bar--bottom': PropTypes.string
        }).isRequired,

        /**
         * An optional `className` to attach to the wrapper.
         */
        className: PropTypes.string,

        /**
         * The contents to be rendered within the `Bar`.
         */
        children: PropTypes.any.isRequired
    };

    render() {
        const {position, className, theme, children, ...rest} = this.props;
        const finalClassName = mergeClassNames({
            [className]: className && className.length,
            [theme.bar]: true,
            [theme['bar--top']]: position === 'top',
            [theme['bar--bottom']]: position === 'bottom'
        });

        return (
            <div className={finalClassName} {...rest}>
                {children}
            </div>
        );
    }
}

export default Bar;
