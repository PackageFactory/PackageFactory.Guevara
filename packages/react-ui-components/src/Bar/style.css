.bar {
    composes: reset from './../reset.css';
    position: absolute;
    left: 0;
    z-index: var(--zIndex-CheckboxInput-Context);
    width: 100%;
    background-color: var(--brandColorsContrastDarker);
    height: 41px;
    border-bottom: 1px solid var(--brandColorsContrastDark);
}
.bar--top {
    top: 0;
}
.bar--bottom {
    bottom: 0;
}
