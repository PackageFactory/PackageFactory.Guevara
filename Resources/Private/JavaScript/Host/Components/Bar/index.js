import React, {PropTypes} from 'react';
import mergeClassNames from 'classnames';
import style from './style.css';

const onDropHandler = (e, cb) => {
    if (cb) {
        e.stopPropagation();

        cb(e);
    }
};

const Bar = props => {
    const {position, className, onDrop} = props;
    const classNames = mergeClassNames({
        [className]: className && className.length,
        [style.bar]: true,
        [style['bar--top']]: position === 'top',
        [style['bar--bottom']]: position === 'bottom'
    });

    return (
        <div
            className={classNames}
            onDragOver={e => e.preventDefault()}
            onDrop={e => onDropHandler(e, onDrop)}
            >
          {props.children}
        </div>
    );
};
Bar.propTypes = {
    // Style related propTypes.
    position: PropTypes.oneOf(['top', 'bottom']).isRequired,
    className: PropTypes.string,

    // Contents of the Bar.
    children: PropTypes.node.isRequired,

    // Interaction related propTypes.
    onDrop: PropTypes.func
};

export default Bar;
