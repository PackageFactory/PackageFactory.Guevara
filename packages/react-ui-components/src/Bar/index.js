import {themr} from '@friendsofreactjs/react-css-themr';
import identifiers from './../identifiers';
import style from './style.css';
import Bar from './bar.js';

export default themr(identifiers.bar, style)(Bar);
