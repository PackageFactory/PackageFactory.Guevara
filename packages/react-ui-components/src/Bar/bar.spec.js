import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Bar from './bar.js';

describe('<Bar/>', () => {
    let props;

    beforeEach(() => {
        props = {
            theme: {},
            children: 'Foo children',
            position: 'top'
        };
    });

    it('should render correctly.', () => {
        const wrapper = shallow(<Bar {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow the propagation of "className" with the "className" prop.', () => {
        const wrapper = shallow(<Bar {...props} className="fooClassName"/>);

        expect(wrapper.prop('className')).toContain('fooClassName');
    });

    it('should allow the propagation of additional props to the wrapper.', () => {
        const wrapper = shallow(<Bar {...props} foo="bar"/>);

        expect(wrapper.prop('foo')).toBe('bar');
    });
});
