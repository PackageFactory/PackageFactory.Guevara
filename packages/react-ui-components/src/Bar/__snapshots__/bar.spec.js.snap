// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar/> should render correctly. 1`] = `
<div
  className=""
>
  Foo children
</div>
`;
